app/views/listings/show.html.erb

```diff
@@ -95,6 +95,9 @@
                 <h4> <%= @listing.user.first_name %> </h4>
             </div>
             <div class="col-sm-1 col-xs-1">
+              <% if current_user.present? %>
+              <%= link_to "Message #{@listing.user.first_name}", conversations_path(initiator_id: current_user.id, recipient_id: @listing.user_id), method: :post %>
+              <% end %>
```

Comment:
So cool! I love how you're linking to this from here, great thought in terms of UX/UI